api/graphql/types/application_section/permissions.py

```diff
@@ -35,3 +36,17 @@ def has_update_permission(cls, instance: ApplicationSection, user: AnyUser, inpu
     @classmethod
     def has_delete_permission(cls, instance: ApplicationSection, user: AnyUser, input_data: dict[str, Any]) -> bool:
         return can_modify_application(user, instance.application)
+
+
+class UpdateAllSectionOptionsPermission(BasePermission):
+    @classmethod
+    def has_update_permission(cls, instance: ApplicationSection, user: AnyUser, input_data: dict[str, Any]) -> bool:
+        if user.is_anonymous:
+            return False
+        if user.is_superuser:
+            return True
+        if has_general_permission(user, required_permission="can_handle_applications"):
+            return True
+
+        units = list(instance.reservation_unit_options.all().values_list("reservation_unit__unit__id", flat=True))
+        return has_unit_permission(user, "can_handle_applications", units)
```

Comment:
Same as application level, does not check that the permission covers all of the (reservation) units.

Reply:
Fixed this and added tests.